source/common/network/socket_option_factory.cc

```diff
@@ -97,8 +97,8 @@ SocketOptionFactory::buildTcpFastOpenOptions(uint32_t queue_length) {
 std::unique_ptr<Socket::Options> SocketOptionFactory::buildIpPacketInfoOptions() {
   std::unique_ptr<Socket::Options> options = std::make_unique<Socket::Options>();
   options->push_back(std::make_shared<AddrFamilyAwareSocketOptionImpl>(
-      envoy::api::v2::core::SocketOption::STATE_BOUND, ENVOY_RECV_IP_PKT_INFO,
-      ENVOY_RECV_IPV6_PKT_INFO, 1));
+      envoy::api::v2::core::SocketOption::STATE_BOUND, ENVOY_IP_PKT_INFO, ENVOY_RECV_IPV6_PKT_INFO,
```

Comment:
I kinda like the original names because they are symmetric between v4 and v6.

Maybe use ENVOY_SELF_IP_ADDR and ENVOY_SELF_IPV6_ADDR, which are symmetric and can be used by both sending and receiving?